code/datums/components/hud_timer.dm

```diff
@@ -0,0 +1,89 @@
+// Adds a timer that ticks down based on the timer datums
+/datum/component/hud_timer
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+	var/mob/owner
+	var/timer_text
+	// The list of timers that are currently being displayed
+	var/list/timers = list()
+	// The list of signals that will start the timer
+	var/start_signals = list()
+	var/stop_signals = list()
+	// The callback to call using the signals. Should return a timer ID
+	var/callback
+// I wonder if it'd be better to make this into two elements, one for nuke and the other for hive orphaning
```

Comment:
Would love to see getting some feedback on this. This component's probably going to be pretty common so making this into hardcoded elements would improve performance by quite a lot

Reply:
In the scope of this game, ~100 things ticking every 1 second isn't really a big deal. I would suggest creating a new subsystem for them and making it's priority low.

Reply:
I would really rather not add a yet another subsystem if at all possible